### PR TITLE
Histogram example - Python

### PR DESCRIPTION
**Pull request by dominikl**
# What this PR does

- Adds `getHistogram` method to the gateway
- Adds an example to the Python training examples.

# Testing this PR

Check that the training jobs passes.

# Related reading

- Docs PR: https://github.com/openmicroscopy/ome-documentation/pull/1804
- https://trello.com/c/5SjwyaM4/112-histogram-api

### CODE REVIEW

File: examples/Training/python/Raw_Data_Access.py [hunk 1]
```diff
@@ -78,6 +78,11 @@
 for i, p in enumerate(planes):
     print "plane zct:", zct_list[i], " min:", p.min(), " max:", p.max()
 
+# Retrieve a histogram
+# ====================
+# Get a 256 bin histogram for channel 0 and plane z=0/t=0:
+hist = pixels.getHistogram(0, 0, 0, 256)
```
Comment (waxenegger) [inline examples/Training/python/Raw_Data_Access.py]: maybe you had good reason to put the getHistogram method on the PixelsWrapper, I wanted to mention that there is a getHistogram on the ImageWrapper already. Tried this locally:

```
hist = image.getHistogram([0], 256, False, 0, 0)
>>> hist
{0L: [1L, 0L, 0L, 0L, 0L, 0L, 0L, 0L, 0L, 6L, 0L, 0L, 9L, 0L, 51L, 0L, 0L, 131L, 0L, 0L, 430L, 0L, 0L, 967L, 0L, 0L, 2194L, 0L, 4056L, 0L, 0L, 7470L, 0L, 0L, 11290L, 0L, 0L, 16620L, 0L, 21019L, 0L, 0L, 25216L, 0L, 0L, 26803L, 0L, 0L, 27057L, 0L, 0L, 23321L, 0L, 20227L, 0L, 0L, 16262L, 0L, 0L, 12444L, 0L, 0L, 8395L, 0L, 0L, 5897L, 0L, 3813L, 0L, 0L, 2644L, 0L, 0L, 1864L, 0L, 0L, 1311L, 0L, 1016L, 0L, 0L, 771L, 0L, 0L, 673L, 0L, 0L, 562L, 0L, 0L, 520L, 0L, 438L, 0L, 0L, 369L, 0L, 0L, 369L, 0L, 0L, 343L, 0L, 363L, 0L, 0L, 337L, 0L, 0L, 346L, 0L, 0L, 361L, 0L, 0L, 333L, 0L, 310L, 0L, 0L, 349L, 0L, 0L, 384L, 0L, 0L, 381L, 0L, 0L, 351L, 0L, 366L, 0L, 0L, 351L, 0L, 0L, 411L, 0L, 0L, 366L, 0L, 447L, 0L, 0L, 425L, 0L, 0L, 450L, 0L, 0L, 432L, 0L, 0L, 515L, 0L, 542L, 0L, 0L, 510L, 0L, 0L, 561L, 0L, 0L, 505L, 0L, 561L, 0L, 0L, 527L, 0L, 0L, 520L, 0L, 0L, 567L, 0L, 0L, 551L, 0L, 471L, 0L, 0L, 448L, 0L, 0L, 472L, 0L, 0L, 447L, 0L, 0L, 432L, 0L, 401L, 0L, 0L, 400L, 0L, 0L, 320L, 0L, 0L, 318L, 0L, 268L, 0L, 0L, 274L, 0L, 0L, 249L, 0L, 0L, 203L, 0L, 0L, 184L, 0L, 132L, 0L, 0L, 120L, 0L, 0L, 92L, 0L, 0L, 53L, 0L, 50L, 0L, 0L, 48L, 0L, 0L, 30L, 0L, 0L, 19L, 0L, 0L, 10L, 0L, 10L, 0L, 0L, 3L, 0L, 0L, 7L, 0L, 0L, 2L, 0L]}

```

**Comment by waxenegger** [comment]
Apart from the one comment about the existing getHistogram method on the ImageWrapper the code in this PR seems to work fine.

```
>>> execfile('Raw_Data_Access.py')

Plane at zct:  0 0 0
[[124 124 117 ..., 119 122 114]
 [119 120 124 ..., 117 120 121]
 [119 121 122 ..., 121 112 123]
 ..., 
 [120 128 121 ..., 117 117 119]
 [123 126 120 ..., 114 112 114]
 [126 121 124 ..., 115 123 118]]
shape:  (512, 512)
min: 104  max: 195 pixel type: uint16

Z stack of tiles:
Tile: (0, 0, 0, (50, 50, 10, 10))  min: 112  max: 129  sum: 12089
Tile: (1, 0, 0, (50, 50, 10, 10))  min: 112  max: 129  sum: 12018
Tile: (2, 0, 0, (50, 50, 10, 10))  min: 113  max: 126  sum: 12020
Tile: (3, 0, 0, (50, 50, 10, 10))  min: 113  max: 126  sum: 12040
Tile: (4, 0, 0, (50, 50, 10, 10))  min: 114  max: 126  sum: 12013
Tile: (5, 0, 0, (50, 50, 10, 10))  min: 113  max: 128  sum: 12028
Tile: (6, 0, 0, (50, 50, 10, 10))  min: 114  max: 127  sum: 12042
Tile: (7, 0, 0, (50, 50, 10, 10))  min: 114  max: 129  sum: 12078
Tile: (8, 0, 0, (50, 50, 10, 10))  min: 114  max: 127  sum: 12052
Tile: (9, 0, 0, (50, 50, 10, 10))  min: 113  max: 126  sum: 12030
Tile: (10, 0, 0, (50, 50, 10, 10))  min: 115  max: 126  sum: 11990
Tile: (11, 0, 0, (50, 50, 10, 10))  min: 115  max: 128  sum: 12034
Tile: (12, 0, 0, (50, 50, 10, 10))  min: 114  max: 130  sum: 12044
Tile: (13, 0, 0, (50, 50, 10, 10))  min: 113  max: 127  sum: 12031
Tile: (14, 0, 0, (50, 50, 10, 10))  min: 113  max: 128  sum: 12052
Tile: (15, 0, 0, (50, 50, 10, 10))  min: 112  max: 129  sum: 12058
Tile: (16, 0, 0, (50, 50, 10, 10))  min: 110  max: 127  sum: 12001
Tile: (17, 0, 0, (50, 50, 10, 10))  min: 113  max: 128  sum: 12021
Tile: (18, 0, 0, (50, 50, 10, 10))  min: 112  max: 129  sum: 12075
Tile: (19, 0, 0, (50, 50, 10, 10))  min: 114  max: 129  sum: 12041
Tile: (20, 0, 0, (50, 50, 10, 10))  min: 109  max: 129  sum: 12041
Tile: (21, 0, 0, (50, 50, 10, 10))  min: 112  max: 129  sum: 12085
Tile: (22, 0, 0, (50, 50, 10, 10))  min: 114  max: 131  sum: 12104
Tile: (23, 0, 0, (50, 50, 10, 10))  min: 113  max: 128  sum: 12118
Tile: (24, 0, 0, (50, 50, 10, 10))  min: 114  max: 127  sum: 12073
Tile: (25, 0, 0, (50, 50, 10, 10))  min: 110  max: 127  sum: 12060
Tile: (26, 0, 0, (50, 50, 10, 10))  min: 114  max: 129  sum: 12084
Tile: (27, 0, 0, (50, 50, 10, 10))  min: 112  max: 128  sum: 12101
Tile: (28, 0, 0, (50, 50, 10, 10))  min: 114  max: 131  sum: 12072

Hyper stack of planes:
plane zct: (14, 0, 0)  min: 105  max: 189
plane zct: (14, 1, 0)  min: 97  max: 122
plane zct: (15, 0, 0)  min: 106  max: 190
plane zct: (15, 1, 0)  min: 96  max: 121
plane zct: (16, 0, 0)  min: 106  max: 189
plane zct: (16, 1, 0)  min: 99  max: 123
plane zct: (17, 0, 0)  min: 105  max: 189
plane zct: (17, 1, 0)  min: 98  max: 122
plane zct: (18, 0, 0)  min: 106  max: 187
plane zct: (18, 1, 0)  min: 98  max: 121
plane zct: (19, 0, 0)  min: 105  max: 188
plane zct: (19, 1, 0)  min: 97  max: 122
plane zct: (20, 0, 0)  min: 105  max: 190
plane zct: (20, 1, 0)  min: 98  max: 122
plane zct: (21, 0, 0)  min: 106  max: 186
plane zct: (21, 1, 0)  min: 98  max: 122
plane zct: (22, 0, 0)  min: 106  max: 186
plane zct: (22, 1, 0)  min: 96  max: 121
plane zct: (23, 0, 0)  min: 105  max: 186
plane zct: (23, 1, 0)  min: 98  max: 122
plane zct: (24, 0, 0)  min: 105  max: 185
plane zct: (24, 1, 0)  min: 99  max: 121
plane zct: (25, 0, 0)  min: 106  max: 182
plane zct: (25, 1, 0)  min: 98  max: 121
plane zct: (26, 0, 0)  min: 106  max: 183
plane zct: (26, 1, 0)  min: 98  max: 122
plane zct: (27, 0, 0)  min: 106  max: 186
plane zct: (27, 1, 0)  min: 98  max: 122
plane zct: (28, 0, 0)  min: 106  max: 181
plane zct: (28, 1, 0)  min: 98  max: 121
[1L, 0L, 0L, 0L, 0L, 0L, 0L, 0L, 0L, 6L, 0L, 0L, 9L, 0L, 51L, 0L, 0L, 131L, 0L, 0L, 430L, 0L, 0L, 967L, 0L, 0L, 2194L, 0L, 4056L, 0L, 0L, 7470L, 0L, 0L, 11290L, 0L, 0L, 16620L, 0L, 21019L, 0L, 0L, 25216L, 0L, 0L, 26803L, 0L, 0L, 27057L, 0L, 0L, 23321L, 0L, 20227L, 0L, 0L, 16262L, 0L, 0L, 12444L, 0L, 0L, 8395L, 0L, 0L, 5897L, 0L, 3813L, 0L, 0L, 2644L, 0L, 0L, 1864L, 0L, 0L, 1311L, 0L, 1016L, 0L, 0L, 771L, 0L, 0L, 673L, 0L, 0L, 562L, 0L, 0L, 520L, 0L, 438L, 0L, 0L, 369L, 0L, 0L, 369L, 0L, 0L, 343L, 0L, 363L, 0L, 0L, 337L, 0L, 0L, 346L, 0L, 0L, 361L, 0L, 0L, 333L, 0L, 310L, 0L, 0L, 349L, 0L, 0L, 384L, 0L, 0L, 381L, 0L, 0L, 351L, 0L, 366L, 0L, 0L, 351L, 0L, 0L, 411L, 0L, 0L, 366L, 0L, 447L, 0L, 0L, 425L, 0L, 0L, 450L, 0L, 0L, 432L, 0L, 0L, 515L, 0L, 542L, 0L, 0L, 510L, 0L, 0L, 561L, 0L, 0L, 505L, 0L, 561L, 0L, 0L, 527L, 0L, 0L, 520L, 0L, 0L, 567L, 0L, 0L, 551L, 0L, 471L, 0L, 0L, 448L, 0L, 0L, 472L, 0L, 0L, 447L, 0L, 0L, 432L, 0L, 401L, 0L, 0L, 400L, 0L, 0L, 320L, 0L, 0L, 318L, 0L, 268L, 0L, 0L, 274L, 0L, 0L, 249L, 0L, 0L, 203L, 0L, 0L, 184L, 0L, 132L, 0L, 0L, 120L, 0L, 0L, 92L, 0L, 0L, 53L, 0L, 50L, 0L, 0L, 48L, 0L, 0L, 30L, 0L, 0L, 19L, 0L, 0L, 10L, 0L, 10L, 0L, 0L, 3L, 0L, 0L, 7L, 0L, 0L, 2L, 0L]

```

**Comment by dominikl** [comment]
Thanks @waxenegger . I missed the `getHistogram` method in the `ImageWrapper`. Will adjust the example accordingly and remove the method from the `PixelWrapper` again.

**Comment by waxenegger** [comment]
That's working and good to be merged.

**Pull request merged**